PopPUNK/__main__.py

```diff
@@ -501,6 +520,34 @@ def main():
 
         nx.write_gpickle(genomeNetwork, args.output + "/" + os.path.basename(args.output) + '_graph.gpickle')
 
+    #******************************#
+    #*                            *#
+    #* within-strain analysis     *#
+    #*                            *#
+    #******************************#
+    sys.stderr.write("Checking options here\n\n")
+    if args.lineage_clustering:
+        sys.stderr.write("Mode: Identifying lineages within a clade\n\n")
+
+        # process rankings to use
+        rank_list = sorted([int(x) for x in args.R.split(',')])
+
+        # load distances
+        distances = None
+        if args.distances is not None:
+            distances = args.distances
+        else:
+            sys.stderr.write("Need to provide an input set of distances with --distances\n\n")
+            sys.exit(1)
+
+        refList, queryList, self, distMat = readPickle(distances)
+        
+        # run lineage clustering
```

Comment:
Perhaps more checking of cmd line args?

Reply:
Added in 8e0875ff87660230a3b41e195d93e3c2f49464a9.